worker_types/gwci-linux/userdata

```diff
@@ -46,22 +47,32 @@ sleep 5
 systemctl status docker | grep "Started Docker Application Container Engine"
 usermod -aG docker ubuntu
 
-# install generic-worker into /home/ubuntu/generic-worker
-mkdir -p /home/ubuntu/generic-worker
-cd /home/ubuntu/generic-worker
-retry curl -L "https://github.com/taskcluster/generic-worker/releases/download/${GENERIC_WORKER_VERSION}/generic-worker-simple-linux-amd64" > generic-worker
+cd /usr/local/bin
+retry curl -L "https://github.com/taskcluster/generic-worker/releases/download/${GENERIC_WORKER_VERSION}/generic-worker-multiuser-linux-amd64" > generic-worker
 retry curl -L "https://github.com/taskcluster/livelog/releases/download/${LIVELOG_VERSION}/livelog-linux-amd64" > livelog
 retry curl -L "https://github.com/taskcluster/taskcluster-proxy/releases/download/${TASKCLUSTER_PROXY_VERSION}/taskcluster-proxy-linux-amd64" > taskcluster-proxy
 chmod a+x generic-worker taskcluster-proxy livelog
-chown -R ubuntu:ubuntu /home/ubuntu/generic-worker
+
+mkdir -p /etc/generic-worker
+mkdir -p /var/local/generic-worker
 ./generic-worker --version
-./generic-worker new-ed25519-keypair --file ed25519.key
+./generic-worker new-ed25519-keypair --file /etc/generic-worker/ed25519_key
 
 # ensure host 'taskcluster' resolves to localhost
 echo 127.0.1.1 taskcluster >> /etc/hosts
-
 # configure generic-worker to run on boot
-echo '@reboot PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin /home/ubuntu/generic-worker/generic-worker run --configure-for-aws --config /home/ubuntu/generic-worker/generic-worker.config >> /home/ubuntu/generic-worker/generic-worker.log 2>&1' | crontab -u ubuntu -
+echo '@reboot cd /var/local/generic-worker && PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin /usr/local/bin/generic-worker run --configure-for-aws --config /etc/generic-worker/config >> /var/log/generic-worker.log 2>&1' | crontab -
+
+retry apt install -y ubuntu-desktop ubuntu-gnome-desktop
+
+# See
+#   * https://console.aws.amazon.com/support/cases#/6410417131/en
+#   * https://bugzilla.mozilla.org/show_bug.cgi?id=1499054#c12
```

Comment:
Interesting, never seen this before.

Reply:
Yeah, I was a little baffled but the AWS support guy was really helpful.